pkg/pillar/zedUpload/datastore_azure.go

```diff
@@ -162,6 +164,31 @@ func (ep *AzureTransportMethod) processAzureUploadByChunks(req *DronaRequest) er
 	return azure.UploadPartByChunk(ep.acName, ep.acKey, ep.container, req.localName, req.UploadID, ep.hClient, req.Adata)
 }
 
+func (ep *AzureTransportMethod) processDownloadByChunks(req *DronaRequest) error {
+	file := req.name
+	dataChan := make(utils.ChunkDataChan)
+	defer close(dataChan)
+	go func(req *DronaRequest, chunkData utils.ChunkDataChan) {
+		for {
+			select {
+			case stats, ok := <-chunkData:
+				if !ok {
+					return
+				}
+				ep.ctx.postChunk(req, stats.Size, stats.Chunk)
+			default:
+				//ignore
```

Comment:
Why are you busy-waiting i.e., spinning around consuming CPU?
The preferred pattern for this is to replace the whole for loop with
   for stats := range chunkData {
		ep.ctx.postChunk(req, stats.Size, stats.Chunk)
   }

And "stats" should really be called "data" since it isn't statistics of any kind.

Reply:
all data will not be available in the chunkData at the given time, as when we have 1MB of data we are writing into channel, that is when chunkData has the byte data to post on the result channel.

renamed to stats